File: Code/Chapter 9/VB.NET/Exercise 9.5/Form1.vb
Imports System.Data

Public Class Form1
   Private myDataSet As CustProd
   Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
      Me.Close()
   End Sub
   Private Sub btnGetChanges_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnGetChanges.Click
      Dim changedDS As CustProd = CType(myDataSet.GetChanges(), CustProd)
      dgCustomers.DataSource = changedDS.Customers
      dgCustomerProducts.DataSource = changedDS.CustomerProducts
      dgProducts.DataSource = changedDS.Products
   End Sub

   Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
      myDataSet = New CustProd()
      dgCustomers.DataSource = myDataSet.Customers
      dgCustomerProducts.DataSource = myDataSet.CustomerProducts
      dgProducts.DataSource = myDataSet.Products

      myDataSet.ReadXml("Data.xml")
      myDataSet.AcceptChanges()
   End Sub
End Class

File: Code/Chapter 9/C#/Exercise 9.5/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Exercise_9_5
{
   partial class Form1 : Form
   {
      private CustProd myDataSet;

      public Form1()
      {
         InitializeComponent();

         myDataSet = new CustProd();
         dgCustomers.DataSource = myDataSet.Customers;
         dgCustomerProducts.DataSource = myDataSet.CustomerProducts;
         dgProducts.DataSource = myDataSet.Products;

         myDataSet.ReadXml("Data.xml");
         myDataSet.AcceptChanges();
      }

      private void btnClose_Click(object sender, EventArgs e)
      {
         this.Close();
      }

      private void btnGetChanges_Click(object sender, EventArgs e)
      {
         CustProd changedDS = (CustProd)myDataSet.GetChanges();
         dgCustomers.DataSource = changedDS.Customers;
         dgCustomerProducts.DataSource = changedDS.CustomerProducts;
         dgProducts.DataSource = changedDS.Products;
      }
   }
}